gapic/templates/setup.py.j2

```diff
@@ -34,7 +34,7 @@ dependencies = [
     {% for package_tuple, package_info in pypi_packages.items() %}
     {# Quick check to make sure the package is different from this setup.py #}
     {% if api.naming.warehouse_package_name != package_info.package_name %}
-    {% if api.requires_package(package_tuple) %}
+    {% if api.requires_package(package_tuple) or (api.has_iam_mixin and package_tuple == ('google', 'iam', 'v1')) %}
```

Comment:
I think google.iam.v1 is special because all other mixins are hosted in api-common-protos package which is a common install.

Reply:
I can move the logic into `requires_package()` as technically `google.iam.v1` is required.

Reply:
Fixed in https://github.com/googleapis/gapic-generator-python/pull/1631/commits/b3307ded78bf33763818d29be62242b0c020ebb0

We don't need to check operations/location mixins because those are part of googleapis-common-protos, and we get that dependency via google-api-core. https://github.com/googleapis/python-api-core/blob/main/setup.py#L32